packages/es-dev-server/src/middleware/plugin-transform.ts

```diff
@@ -18,12 +52,27 @@ export function createPluginTransformMiddlware(cfg: PluginServeMiddlewareConfig)
   }
 
   return async function pluginTransformMiddleware(context, next) {
+    let cached = false;
+    let cacheKey = '';
+    let cachedBody: string | undefined;
+
+    if (cfg.fileWatcher && cfg.cache) {
+      const result = await tryServeFromCache(cfg.cache, context);
+      ({ cached, cacheKey, cachedBody } = result);
+      context = result.context;
+    }
+
     await next();
 
     if (context.status < 200 || context.status >= 300) {
       return undefined;
     }
 
+    if (cached) {
+      context.body = cachedBody;
```

Comment:
If we're serving from the cache we can avoid running the other middleware. We can just serve it directly, so we can place this above the `await next()`

Reply:
Yep you're right, done :)!